src/persistentstorage.cpp

```diff
@@ -258,3 +266,229 @@ void MysqlPersistentStorage::writelog(const std::string &id, const ObjectValue &
     }
 }
 #endif // HAVE_MYSQL
+
+
+#ifdef HAVE_INFLUXDB
+Logger& InfluxdbPersistentStorage::logger_m(Logger::getInstance("InfluxdbPersistentStorage"));
+
+bool _existsInArray(const Json::Value &json, const std::string &key)
+{
+    bool ret = false;
+    assert(json.isArray());
+    for (int i = 0; i != json.size(); i++)
+    {
+        if (json[i][0].asString() == key)
+            ret = true;
+    }
+    return ret;
+}
+
+bool InfluxdbPersistentStorage::createDB(const std::string &db, std::string &result)
+{
+    std::string query = "q=CREATE DATABASE ";
+    query.append(db);
+    return curlRequest(INFLUXDB_QUERY, db, query, result);
+}
+
+InfluxdbPersistentStorage::InfluxdbPersistentStorage(ticpp::Element* pConfig)
+{
+    uri_m = pConfig->GetAttribute("uri");
+    user_m = pConfig->GetAttribute("user");
+    pass_m = pConfig->GetAttribute("pass");
+    db_m = pConfig->GetAttribute("db");
+    persist_db_m = pConfig->GetAttribute("persist_db");
+
+    std::string resp;
+
+    if (curlRequest(INFLUXDB_QUERY, db_m, "q=SHOW DATABASES", resp))
+    {
+        Json::CharReaderBuilder builder;
+        Json::CharReader * reader = builder.newCharReader();
+        Json::Value root, val;
+        std::string errors;
+
+        if(!reader->parse(resp.c_str(), resp.c_str()+resp.size(), &root, &errors))
+        {
+            logger_m.errorStream() << "couldn't parse JSON! Errors: " << errors << endlog;
+        }
+        else
+        {
+            Json::Value databases;
+            resp.clear();
+            databases = root["results"][0]["series"][0]["values"];
+            if (!_existsInArray(databases, db_m))
+                if (!createDB(db_m, resp))
+                    logger_m.errorStream() << "couldn't create logging database: " << resp << endlog;
+            if (!_existsInArray(databases, persist_db_m))
+                if (!createDB(persist_db_m, resp))
+                    logger_m.errorStream() << "couldn't create persistence database: " << resp << endlog;
+        }
+    }
+}
+
+InfluxdbPersistentStorage::~InfluxdbPersistentStorage()
+{
+}
+
+static size_t _curlWriteCB(void *contents, size_t size, size_t nmemb, void *userp)
+{
+    ((std::string*)userp)->append((char*)contents, size * nmemb);
+    return size * nmemb;
+}
+
+#define INFLUXDB_EXTRA_DEBUG 0
+
+bool InfluxdbPersistentStorage::curlRequest(InfluxdbOperation_t oper, const std::string& db, const std::string& query, std::string& result)
+{
+    bool ret = false;
+    CURL *curl;
+	CURLcode res;
+	curl = curl_easy_init();
+	if(curl)
+	{
+		std::stringstream url;
+        url << uri_m << "/" << (oper == INFLUXDB_WRITE ? "write" : "query") << "?db=" << db;
+        if (!user_m.empty() && !pass_m.empty())
+        {
+            url << "&u=" << user_m << "&p=" << pass_m;
+        }
+        const char *field_data = query.c_str();
```

Comment:
@fraxinas You can get rid of that poorly-named local var and use `query.c_str()` in `curl_easy_setopt` below.